### PR TITLE
Separating out operator unit tests.

### PR DESCRIPTION
**Pull request by mdavis36**
# Summary

- This PR is a refactoring
- It does the following :
  - Splits the operator unit tests into multiple test files to help with compile times on xl.

xl compile times on ansel.
Develop : ~ 4m 24s
PR sequential : ~0m 50s
PR parallel : ~0m 12s

### CODE REVIEW

File: test/unit/util/operator/test-operators-identity-project.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
```
Comment (rhornung67) [inline test/unit/util/operator/test-operators-identity-project.cpp:1]: What does the term "project" in filename and in test names refer to?

File: test/unit/util/operator/test-operators-basic.cpp [hunk 2]
```diff
@@ -0,0 +1,102 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
```
Comment (rhornung67) [inline test/unit/util/operator/test-operators-basic.cpp:1]: Would it be clearer to replace "basic" with "math" in the filename and test names?

**Review by rhornung67 — COMMENTED** [review]
Oveerall looks great, especially the reduced compile times. Just a couple of comments/questions to consider.

File: test/unit/util/operator/test-operators-identity-project.cpp (hunk 1 above)
Comment (mdavis36) [reply test/unit/util/operator/test-operators-identity-project.cpp:1]: the project1st and project2nd operator tests

File: test/unit/util/operator/test-operators-basic.cpp (hunk 2 above)
Comment (mdavis36) [reply test/unit/util/operator/test-operators-basic.cpp:1]: Sure, will do

**Comment by mdavis36** [comment]
LGTM

File: test/unit/util/operator/test-operators-identity-project.cpp (hunk 1 above)
Comment (rhornung67) [reply test/unit/util/operator/test-operators-identity-project.cpp:1]: I still don't understand.

File: test/unit/util/operator/test-operators-identity-project.cpp [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  if (std::is_signed<T>::value) {
+    j = static_cast<T>(-1);
+    ASSERT_EQ(id(j), T(-1));
+  }
+}
+
+template<typename T>
+void project1st_test()
+{
+  using Proj1 = RAJA::operators::project1st<T, T>;
```
Comment (mdavis36) [inline test/unit/util/operator/test-operators-identity-project.cpp:40]: @rhornung67 The file and test names refer to this operator type and the one used on line 58

File: test/unit/util/operator/test-operators-identity-project.cpp (hunk 1 above)
Comment (artv3) [reply test/unit/util/operator/test-operators-identity-project.cpp:1]: @mdavis36 , do you mean like a projection?

**Review by rhornung67 — APPROVED** [review]
Looks good @mdavis36 thanks!

**Comment by mdavis36** [comment]
LGTM

**Pull request merged**